umh-core/pkg/service/redpanda/redpanda.go

```diff
@@ -0,0 +1,1010 @@
+		metrics.Cluster.Topics = getMetricValue(family.Metric[0])
+	}
+
+	if family, ok := mf["redpanda_cluster_unavailable_partitions"]; ok && len(family.Metric) > 0 {
+		metrics.Cluster.UnavailableTopics = getMetricValue(family.Metric[0])
+	}
+
+	// Throughput metrics
+	if family, ok := mf["redpanda_kafka_request_bytes_total"]; ok {
+		// Process only produce/consume metrics in a single pass
+		for _, metric := range family.Metric {
+			if label := getLabel(metric, "redpanda_request"); label != "" {
+				if label == "produce" {
+					metrics.Throughput.BytesIn = getMetricValue(metric)
+				} else if label == "consume" {
+					metrics.Throughput.BytesOut = getMetricValue(metric)
+				}
+			}
+		}
+	}
+
+	// Topic metrics
+	if family, ok := mf["redpanda_kafka_partitions"]; ok {
+		for _, metric := range family.Metric {
+			if topic := getLabel(metric, "redpanda_topic"); topic != "" {
+				metrics.Topic.TopicPartitionMap[topic] = getMetricValue(metric)
+			}
+		}
+	}
+
+	return metrics, nil
+}
+
+// getMetricValue extracts numeric value from a metric
+func getMetricValue(m *dto.Metric) int64 {
+	if m.Counter != nil {
+		return int64(m.Counter.GetValue())
+	}
+	if m.Gauge != nil {
+		return int64(m.Gauge.GetValue())
+	}
+	if m.Untyped != nil {
+		return int64(m.Untyped.GetValue())
+	}
+	return 0
+}
+
+// getLabel extracts a label value from a metric
+func getLabel(m *dto.Metric, name string) string {
+	for _, label := range m.Label {
+		if label.GetName() == name {
+			return label.GetValue()
+		}
+	}
+	return ""
+}
+
+// GetHealthCheckAndMetrics returns the health check and metrics of a Redpanda service
+func (s *RedpandaService) GetHealthCheckAndMetrics(ctx context.Context, tick uint64, logs []s6service.LogEntry) (RedpandaStatus, error) {
+	start := time.Now()
+	defer func() {
+		metrics.ObserveReconcileTime(logger.ComponentRedpandaService, constants.RedpandaServiceName, time.Since(start))
+	}()
+
+	if ctx.Err() != nil {
+		return RedpandaStatus{}, ctx.Err()
+	}
+
+	// Skip health checks and metrics if the service doesn't exist yet
+	// This avoids unnecessary errors in Status() when the service is still being created
+	if _, exists := s.s6Manager.GetInstance(constants.RedpandaServiceName); !exists {
+		return RedpandaStatus{
+			HealthCheck: HealthCheck{
+				IsLive:  false,
+				IsReady: false,
+			},
+			Metrics: Metrics{},
+			Logs:    []s6service.LogEntry{},
+		}, nil
+	}
+
+	baseURL := "http://localhost:9644"
+	metricsEndpoint := baseURL + "/public_metrics"
+
+	// Create a client to use for our requests
+	// If it's a mock client (used in tests), use it directly
+	// Otherwise, create a new client with timeouts based on context
+	var requestClient httpclient.HTTPClient = s.httpClient
+
+	// Only create a default client if we're not using a mock client
+	if requestClient == nil {
+		requestClient = httpclient.NewDefaultHTTPClient()
+	}
+
+	// Check for ready state using log search (move this to a separate goroutine)
```

Comment:
why put the log search here in the "GetHealthCheckAndMetrics" routine? in other packages, we only check here the stuff from the HTTP endpoints. Maybe it is better suited during the startup procedure (as it also is for the other fsms)

Reply:
Because there is no good way to get the actual redpanda readiness from just calling HTTP endpoints.